server/index.js

```diff
@@ -72,6 +68,15 @@ postload.forEach((middleware) => app.use(middleware))
 
 // error handler for rendering the 404 and 500 pages, must go last
 app.use(errorPages)
-app.listen(parseInt(process.env.PORT || '3000', 10))
 
-module.exports = app
+// instantiate a top-level express instance that will
+//  1. handle static asset requests, skipping auth and other non-CSP middleware
+//  2. delegate all other requests to the main app
+const server = express()
```

Comment:
Typically to prevent authentication logic from affecting assets served I'd just attach the static middleware higher up in this file (before userAuth is applied). Can you explain the considerations in taking this approach instead?

Reply:
In production, I often configure nginx to serve static assets directly, bypassing the JS process altogether. I had that architecture in mind when I wrote this — but you're right, just attaching the middleware  earlier would be better!